### DISCUSSION TITLE
Expand multiple items, manipulating each item first

### DISCUSSION THREAD

**Discussion by rieje** [junegunn/fzf#4153] [Q&A] [answered]
I'm implementing `ctrl-d` to trash items:

    selection=$(fd . "$HOME/.config/alacritty/themes/" --format '{/}' |
      sed 's/.toml$//' | SHELL=/usr/bin/bash fzf \
      --header "original theme: $theme" \
      --preview='set_theme {}' \
      --bind "ctrl-d:execute-silent(trash "$alacritty_theme_dir/{+}.toml")")

example of items: `aura`, `breeze`, `dracula` (names of themes). When selected, I want to trash them as `~/.config/alacritty/themes/<theme>.toml`. When these names of themes are selected, can they be manipulated to be their actual paths then trashed? 

With `{+}` as in above, it will expand into something like `trash "$alacritty_theme_dir/'aura' 'breeze' 'dracula'.toml"`

Any ideas?

Much appreciated.

**Comment by LangLangBart** [comment, marked as answer]
You could try to take advantage of `fd`'s `--format` flag by adding the listed values twice: once as `{/.}`, the basename without the file extension, and once as `{}`, the normal path, separated by a character you like. In the example below, the `Record separator` `\x1e`[^1] [^2] is used.

Then, pipe the output into `fzf`, set the `delimiter` to your chosen character, and control what is displayed with `--with-nth`. After this, you can use `{1}` or `{2}` to perform the desired action.

```bash
fd -a --format $'{/.}\x1e{}' | fzf --delimiter '\x1e' --with-nth 1 --preview 'echo {+2}'
```

<img src="https://github.com/user-attachments/assets/f5ccecb1-fd3e-49a4-bb4d-136bb430550e" width="400">



[^1]: [ASCII Table -- Nonprintable Characters](https://condor.depaul.edu/sjost/lsp121/documents/ascii-npr.htm)
[^2]: [RFC 20 - ASCII format for network interchange](https://datatracker.ietf.org/doc/html/rfc20#section-4.1)

**Reply by rieje** [reply]
Works nicely. I am having an unrelated problem with reloading:

    if selection=$(fd -a . "$HOME/.config/alacritty/themes/" --format $'{/.}\x1e{}' |
      SHELL=/usr/bin/bash fzf \
      --delimiter '\x1e' --with-nth 1 \
      --header "original theme: $theme" \
      --preview='set_theme {+2}' \
      --bind "ctrl-d:execute-silent(trash '{+2}')+reload(fd -a . "$HOME/.config/alacritty/themes/" --format $'{/.}\x1e{}')"); then
      set_theme permanent "${selection##*$'\x1e'}"
    else
      set_theme "$alacritty_theme_dir/$theme.toml"
    fi

When I `ctrl-d`, it `trash`es the file as expected but also appears to apply `set_theme`. My intention is to just trash the file and re-run `fd`/`fzf` to update the list and continue previewing and potentially `ctrl-d` to trash again.

Any ideas?

Thanks.

**Reply by LangLangBart** [reply]
I guess the issue here is proper quoting and escaping '{}' as it's special when 'fzf' sees it.

Try assigning your command to a variable, like this.

```bash
fd_command=$'fd -a . $HOME/.config --format {/.}\x1e\\{}'
fzf \
  --bind "start:reload:$fd_command" \
  --bind "ctrl-d:execute(echo hi)+reload:$fd_command" \
  --delimiter '\x1e' \
  --preview 'echo {2}' \
  --with-nth 1
```

PS: this will also be executed `--preview='set_theme {+2}'`, make sure its intentional
